Mathlib/Algebra/Algebra/Bilinear.lean

```diff
@@ -25,16 +24,6 @@ section NonUnitalNonAssoc
 variable (R A : Type*) [CommSemiring R] [NonUnitalNonAssocSemiring A] [Module R A]
   [SMulCommClass R A A] [IsScalarTower R A A]
 
-/-- The multiplication in a non-unital non-associative algebra is a bilinear map.
-
-A weaker version of this for semirings exists as `AddMonoidHom.mul`. -/
-def mul : A →ₗ[R] A →ₗ[R] A :=
-  LinearMap.mk₂ R (· * ·) add_mul smul_mul_assoc mul_add mul_smul_comm
```

Comment:
I'm not sure I understand what is gained by moving this earlier, this definitely feels like it should remain with everything except the tensor stuff up to line 90.

Moving `mulLeft` and `mulRight` and `mulLeftRight` earlier too would be fine.

Reply:
Great! I'll (try to) review that PR to make up for dropping this PR :)